AU2/plugins/util/WantedManager.py

```diff
@@ -1,67 +1,63 @@
 
     def add_event(self, e: Event):
         for playerID in e.pluginState.get("WantedPlugin", {}):
             duration, crime, redemption = e.pluginState["WantedPlugin"][playerID]
-            expiry = e.datetime + datetime.timedelta(days=duration)
             self.wanted_events.setdefault(playerID, [])
-            self.wanted_events[playerID].append((expiry, (duration, crime, redemption)))
-
-    def is_wanted_at(self, a: Assassin, date: datetime.datetime):
-        """
-        Returns true if:
-         1) the WantedPlugin is activated
-         2) `a` has a wanted event before `date`
-         2) `a`'s most recent wanted_event (at `date`) puts them wanted at `date`
-        """
-        if self.activated:
-            if not self.wanted_events.get(a.identifier, False):
-                return False
-            most_recent_event = 0
-            for i in self.wanted_events[a.identifier]:
-                if (i[0] - datetime.timedelta(days=i[1][0])) > date:
-                    break
-                most_recent_event = i
-            if not most_recent_event:
-                return False
-            else:
-                return most_recent_event[0] > date
-        else:
-            return False
-
-    def get_wanted_players(self, date: datetime.datetime):
-        """
-        Returns all players who are wanted at `date`
-        """
-        return [a for a in ASSASSINS_DATABASE.assassins.values() if self.is_wanted_at(a, date)]
+            self.wanted_events[playerID].append({'event_time': e.datetime,
+                                                 'wanted_duration': datetime.timedelta(days=duration),
+                                                 'crime': crime,
+                                                 'redemption': redemption})
+        for (killer, victim) in e.kills:
+            self.wanted_events.setdefault(victim, [])
+            self.wanted_events[victim].append({'event_time': e.datetime})
+
+    def get_live_wanted_players(self, police=False):
+        players = {}
+        for player_id in self.wanted_events:
+            if ASSASSINS_DATABASE.get(player_id).is_police == police:
+                last_event = self.wanted_events[player_id][-1]
+                if len(last_event.keys()) == 1:
+                    # if a players last event is a death, then they haven't committed a post-resurrection crime, so ignore
+                    continue
+                # Check most recent event for wanted info. No support for multiple active wanted_events per player
+                if last_event['event_time'] + last_event['wanted_duration'] >= self.current_time:
+                    players[player_id] = {'crime': last_event['crime'], 'redemption': last_event['redemption']}
+        return players
+
+    def get_wanted_player_deaths(self, police=False):
+        wanted_deaths = []
+        for player_id in self.wanted_events:
+            if ASSASSINS_DATABASE.get(player_id).is_police == police:
```

Comment:
This function is prime for de-nesting using some of the tricks I taught you. 

For example:

```python
if ASSASSINS_DATABASE.get(player_id).is_police == police:
    BLOCK
```

can be 

```python
if ASSASSINS_DATABASE.get(player_id).is_police != police:
    continue
BLOCK
```

etc.